symfony/client/src/orangehrmPerformancePlugin/pages/KpiEdit.vue

```diff
@@ -0,0 +1,233 @@
+                label="Minimum Rating"
+                required
+                autcomplete="off"
+                :rules="rules.minRate"
+              />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <oxd-input-field
+                v-model.number="kpi.maxRate"
+                label="Maximum Rating"
+                required
+                autcomplete="off"
+                :rules="rules.maxRate"
+              />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <div class="orangehrm-module-field-row">
+                <oxd-text tag="p" class="orangehrm-module-field-label">
+                  Make Default Scale
+                </oxd-text>
+                <oxd-switch-input v-model="kpi.isDefault" />
+              </div>
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+
+        <oxd-divider />
+        <br />
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button
+            type="button"
+            display-type="ghost"
+            label="Cancel"
+            @click="onCancel"
+          />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import SwitchInput from '@ohrm/oxd/core/components/Input/SwitchInput';
+import {navigate} from '@ohrm/core/util/helper/navigation';
+import {
+  required,
+  digitsOnly,
+  shouldNotExceedCharLength,
+  minValueShouldBeLowerThanMaxValue,
+} from '@ohrm/core/util/validation/rules';
+import {APIService} from '@/core/util/services/api.service';
+import JobtitleDropdown from '@/orangehrmPerformancePlugin/components/JobtitleDropdown.vue';
+
+const initialKpi = {
+  title: '',
+  jobTitle: null,
+  minRate: null,
+  maxRate: null,
+  isDefault: false,
+};
+
+const rateError = v => {
+  if (digitsOnly(v)) {
+    return (
+      (v >= 0 && v <= 100 && Math.round(v) === v) ||
+      'Should be a number between 0-100'
+    );
+  }
+  return 'Should be a number between 0-100';
+};
+
+export default {
+  name: 'KpiSave',
+  components: {
+    'oxd-switch-input': SwitchInput,
+    'jobtitle-dropdown': JobtitleDropdown,
+  },
+  props: {
+    kpiId: {
+      type: String,
+      required: true,
+    },
+  },
+  setup() {
+    const http = new APIService(
+      'https://02594277-e771-4db3-a5ec-ad645ec49c02.mock.pstmn.io',
+      'api/v2/performance/kpi',
+    );
+
+    return {
+      http,
+    };
+  },
+  data() {
+    return {
+      isLoading: false,
+      kpi: {...initialKpi},
+      rules: {
+        title: [required, shouldNotExceedCharLength(100)],
+        jobTitle: [required],
+        minRate: [
+          required,
+          rateError,
+          v =>
+            !this.kpi.maxRate ||
+            v <= this.kpi.maxRate ||
+            'Minimum Rating should be less than Maximum Rating',
+        ],
+        maxRate: [
+          required,
+          rateError,
+          minValueShouldBeLowerThanMaxValue(
```

Comment:
I think we should rename this function as _**maxValue**ShouldBeGreaterThanMinValue_

 _**minValue**ShouldBeLowerThanMaxValue_ is grammatically correct but sound wrong to be used in max value field.

Reply:
Let's refactor later. Added this comment as a reminder.